### PR TITLE
Use NDK from ANDROID_NDK_HOME + Update NDK to r21e

### LINKED ISSUE #935: Support specifying the NDK location using an env var
Right now we always require it to be in 
https://github.com/evilcorpltd/aTox/blob/1c4d05637a9c242e67b7b9326cac14eb91ac16cc/scripts/android.mk#L1
which is pretty silly, since I already have an NDK elsewhere for e.g. the Bazel build. It would be nice to support using the same NDK for the from-source build. `ATOX_NDK_HOME`? Reusing `ANDROID_NDK_HOME`?

### PR DESCRIPTION
**Pull request by roihershberg**
Resolves #935.

### CODE REVIEW

File: scripts/android.mk [hunk 1]
```diff
@@ -1,7 +1,7 @@
-NDK_HOME	:= $(SRCDIR)/$(NDK_DIR)
+ANDROID_NDK_HOME ?= $(SRCDIR)/$(NDK_DIR)
```
Comment (robinlinden) [inline scripts/android.mk:1]: How about
```
ANDROID_NDK_HOME ?= $(SRCDIR)/$(NDK_DIR)
NDK_HOME := ANDROID_NDK_HOME
```
?

That way we can also pretty easily chain in `ANDROID_NDK_ROOT` without changing all the variable names again if we want to support the `ANDROID_NDK_ROOT/21.lots.of.numbers` in the future as well.

File: scripts/android.mk [hunk 2]
```diff
@@ -56,4 +49,12 @@ $(TOOLCHAIN_FILE): $(NDK_HOME) scripts/android.mk
 	echo 'set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)' >> $@
 	@$(POST_RULE)
 
+$(ANDROID_NDK_HOME):
```
Comment (robinlinden) [inline scripts/android.mk:52]: Why move this block down here? The rule above depends (sometimes!) on this rule, so being able to read it from top to bottom is pretty nice, and it seems like an unrelated change?

Comment (roihershberg) [reply scripts/android.mk:52]: I can move it back up. It's from my searches about how to check if directory exists and as I didn't know if ordering matters I just did like the example.

File: scripts/android.mk (hunk 1 above)
Comment (roihershberg) [reply scripts/android.mk:1]: OK

File: scripts/android.mk (hunk 2 above)
Comment (robinlinden) [reply scripts/android.mk:52]: Ah, right. Yeah, I don't think the order of the rules should matter in general. In almost all cases, they're just looked up as-needed, so `./scripts/build-aarch64-linux-android release` will say we require `release`, then that sort of cascades..
`release: _build/$(TARGET)/tox4j/libtox4j-c$(DLLEXT)` ->
`_build/$(TARGET)/tox4j/libtox4j-c$(DLLEXT): $(PREFIX)/tox4j.stamp` ->
`$(PREFIX)/tox4j.stamp: $(BUILDDIR)/tox4j/Makefile` -> etc, until it reaches something that depends on ANDROID_NDK_HOME.

Comment (roihershberg) [reply scripts/android.mk:52]: Yeah it shouldn't matter,

**Review by robinlinden — APPROVED** [review]
Looks and works well! Tested locally both with ANDROID_NDK_HOME set and without it. :) Thank you!

**Comment by roihershberg** [comment]
Nice!
